### PR TITLE
Generative rotated molecules from a finite set of particules

### PR DESCRIPTION
**Pull request by NicolasLegendre1**

### CODE REVIEW

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:21]: number -> Number

Comment (ninamiolane) [reply sim_volumes.py:21]: And same with all parameters descriptions in `Parameters` and `Returns`:
description starts with uppercase and ends with a .

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:17]: How do you know this samples from a __uniform__ distribution, do you have a reference (Wikipedia, research paper) that explains why these formulas work?

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:30]: Would:
`u1, u2, u3 = np.random.rand(3, num_pts)` work?

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:15]: Let's call this:
`uniform_quaternion` to be consistent with the naming of the next function.

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:58]: remove leftover comment

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:166]: add spaces before the , 

Is this line missing >>> ?

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:169]: does this work if `main_dir` doesn't end with a `/`?

To avoid this problem, you can use:
os.path.join(main_dir, name + "_molecules.npy") to create the path.

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:132]: Why do we want to convert to string? it might not seem natural from a user perspective that this function returns a string of a quaternion. Is it possible to just return the quaternion, and convert it to string whenever it's really needed?

File: sim_volumes.py
Comment (ninamiolane) [inline sim_volumes.py:64]: Can you explain what "modify weight" means? I.e. which weights are modified and why?

File: test_sim_volumes.py
Comment (ninamiolane) [inline test_sim_volumes.py:32]: How did you get this value?

If you run `sim_volumes.modify_weight` on your computer and then copy-pasted it here, it's slightly less effective as a unit-test (it's still useful, as it protects your code from being modified by other contributors).

Is there a simpler test (with a simpler volume) that could be run here, so that the volume[1][1][1] can be computed manually - so that the manual computation is the check for the `modify_weight`'s computation?

**Review by ninamiolane — COMMENTED** [review]
Looking good!! Thanks @NicolasLegendre1 . I've left a few comments.

**Comment by ninamiolane** [comment]
@NicolasLegendre1 you can ignore the DeepSource errors.

@fredericpoitevin could you make deepsource automatically ignore the above errors (see my comment in PR https://github.com/compSPI/ioSPI/pull/14)? There's no hurry though, we can merge the PRs with DeepSource failing.

Also, it seems that the DeepSource transformers have been run on this PR, as opposed to PR https://github.com/compSPI/ioSPI/pull/14 !

The difference between this PR and the PR#14 from ioSPI is that PR#14 comes from the branch on Nicolas's fork, and this PR is directly on a branch on the main repo. In order for the DeepSource transformers to run on your PR#14, Nicolas you will need to enable DeepSource on your own ioSPI repo.

**Comment by fredericpoitevin** [comment]
> @NicolasLegendre1 you can ignore the DeepSource errors.
> 
> @fredericpoitevin could you make deepsource automatically ignore the above errors (see my comment in PR [compSPI/ioSPI#14](https://github.com/compSPI/ioSPI/pull/14))? There's no hurry though, we can merge the PRs with DeepSource failing.
> 
> Also, it seems that the DeepSource transformers have been run on this PR, as opposed to PR [compSPI/ioSPI#14](https://github.com/compSPI/ioSPI/pull/14) !
> 
> The difference between this PR and the PR#14 from ioSPI is that PR#14 comes from the branch on Nicolas's fork, and this PR is directly on a branch on the main repo. In order for the DeepSource transformers to run on your PR#14, Nicolas you will need to enable DeepSource on your own ioSPI repo.

Thanks @NicolasLegendre1 and @ninamiolane! 
I did add the "ignore rule" but I think we should merge regardless. Once merged, I'll create a `tests/` directory and move the test files there - this way we have the same structure across all compSPI repos and also the same deepsource setup.

**Comment by ninamiolane** [comment]
@fredericpoitevin @NicolasLegendre1 deep sources errors are ignored! Ready to merge once the comments are addressed 🎉

File: sim_volumes.py
Comment (NicolasLegendre1) [reply sim_volumes.py:17]: okay I will add that but i'm not sure how to present that into the Docstring of the function

**Review by ninamiolane — APPROVED** [review]
@NicolasLegendre1 @fredericpoitevin LGTM! (I've seen a few other details, but in the interest of merging PRs slightly faster I am OK to merge now 🤗 ). Should we contact Youssef or Julien to officially merge here?

**Comment by fredericpoitevin** [comment]
> @NicolasLegendre1 @fredericpoitevin LGTM! (I've seen a few other details, but in the interest of merging PRs slightly faster I am OK to merge now 🤗 ). Should we contact Youssef or Julien to officially merge here

Let's merge this PR now. I'd like to homogenize the structure across compSPI repos before soliciting maintainers: what do you think?

**Review by fredericpoitevin — APPROVED** [review]

**Pull request merged**
